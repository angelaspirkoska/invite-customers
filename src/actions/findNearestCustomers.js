export const FIND_NEAREST_CUSTOMER = "FIND_NEAREST_CUSTOMER";

/**
 * action creator findNearestCustomer
 * creates action with type FIND_NEAREST_CUSTOMER and payload value withinKilometers
 * @param {int} withinKilometers max kilometer distance from the office point
 */
export default function findNearestCustomer(withinKilometers) {
  return { type: FIND_NEAREST_CUSTOMER, payload: withinKilometers };
}
