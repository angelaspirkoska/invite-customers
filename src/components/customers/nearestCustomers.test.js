import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import _ from "lodash";

import NearestCustomers from "./nearestCustomers";
import { FIND_NEAREST_CUSTOMER } from "../../actions/findNearestCustomers";

//because we have customer data in a file, we can set up test data to do tests on
import { nearestCustomers } from "../../tests/testData/nearestCustomersTestData";

const mockStore = configureStore([]);

describe("renders nearest customers component correctly ", () => {
  let store;
  let component;
  let action = { type: FIND_NEAREST_CUSTOMER, payload: 100 };
  beforeEach(() => {
    store = mockStore({
      nearestCustomers: nearestCustomers,
    });

    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <NearestCustomers />
      </Provider>
    );
  });

  test("should render with given state from Redux store", () => {
    const trElements = component.root.findAllByType("tr");
    expect(trElements.length).toBe(17);

    _.forEach(trElements, (trElement) => {
      expect(trElement.children.length).toBe(3);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("should dispatch findNearestCustomers action", () => {
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(action);
  });
});
