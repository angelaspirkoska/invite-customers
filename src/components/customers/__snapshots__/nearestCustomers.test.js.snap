// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nearest customers component correctly  should render with given state from Redux store 1`] = `
Array [
  <p
    className="title"
  >
    Nearest customers
  </p>,
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Distance from office
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          4
        </td>
        <td>
          Ian Kehoe
        </td>
        <td>
          11 km
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          Nora Dempsey
        </td>
        <td>
          23 km
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Theresa Enright
        </td>
        <td>
          24 km
        </td>
      </tr>
      <tr>
        <td>
          11
        </td>
        <td>
          Richard Finnegan
        </td>
        <td>
          38 km
        </td>
      </tr>
      <tr>
        <td>
          39
        </td>
        <td>
          Lisa Ahearn
        </td>
        <td>
          38 km
        </td>
      </tr>
      <tr>
        <td>
          12
        </td>
        <td>
          Christina McArdle
        </td>
        <td>
          42 km
        </td>
      </tr>
      <tr>
        <td>
          15
        </td>
        <td>
          Michael Ahearn
        </td>
        <td>
          44 km
        </td>
      </tr>
      <tr>
        <td>
          31
        </td>
        <td>
          Alan Behan
        </td>
        <td>
          44 km
        </td>
      </tr>
      <tr>
        <td>
          13
        </td>
        <td>
          Olive Ahearn
        </td>
        <td>
          62 km
        </td>
      </tr>
      <tr>
        <td>
          29
        </td>
        <td>
          Oliver Ahearn
        </td>
        <td>
          72 km
        </td>
      </tr>
      <tr>
        <td>
          30
        </td>
        <td>
          Nick Enright
        </td>
        <td>
          83 km
        </td>
      </tr>
      <tr>
        <td>
          23
        </td>
        <td>
          Eoin Gallagher
        </td>
        <td>
          83 km
        </td>
      </tr>
      <tr>
        <td>
          8
        </td>
        <td>
          Eoin Ahearn
        </td>
        <td>
          84 km
        </td>
      </tr>
      <tr>
        <td>
          24
        </td>
        <td>
          Rose Enright
        </td>
        <td>
          89 km
        </td>
      </tr>
      <tr>
        <td>
          17
        </td>
        <td>
          Patricia Cahill
        </td>
        <td>
          96 km
        </td>
      </tr>
      <tr>
        <td>
          26
        </td>
        <td>
          Stephen McArdle
        </td>
        <td>
          99 km
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="button-container"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
    >
      Invite customers
    </button>
  </div>,
]
`;
